addons/mrp/models/mrp_production.py

```diff
@@ -687,6 +686,8 @@ def post_inventory(self):
             moves_to_do = order.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
             for move in moves_to_do.filtered(lambda m: m.product_qty == 0.0 and m.quantity_done > 0):
                 move.product_uom_qty = move.quantity_done
+            for line in moves_to_do.mapped('move_line_ids').filtered(lambda l: l.product_id.tracking != 'none' and not l.lot_id):
```

Comment:
you can even unlink them IMO

Reply:
Unfortunately, this would break a flow:
(Suppose the current tab in your explorer is T1)

1. Steps 1 -> 6 in PR description
2. In another tab T2, process the work order
3. Record a production of 3 P_finished (Note that after the recording, the page is redirected back to the work order processing)
4. In T1, Post Inventory
5. In T2, the current producing quantity of P_finished is 2. Change it to 1

Error: "Record does not exist or has been deleted."
When changing the quantity to 1, an `onchange` is triggered. However, the `onchange` values still contain the ID of the deleted SML (in field `active_move_line_ids`). During the `onchange`, this field is used: 
https://github.com/odoo/enterprise/blob/35ad7ebd7b04b96d0c2e5e3b93865487993683c8/mrp_workorder/models/mrp_workorder.py#L82
And since the record does not exist, it will raise an error.

I don't know if other flows could be broken too, so I think it would be better to keep the current fix ?

Reply:
'Bien Vu' they will be deleteed anyway at validation so it's not a big deal
Thanks to have tried this